manim/animation/indication.py

```diff
@@ -247,59 +414,169 @@ def get_rect(self) -> SurroundingRectangle:
+
     def __init__(
         self,
         mobject: "Mobject",
         rect_animation: Animation = ShowCreationThenDestruction,
         **kwargs
     ) -> None:
+        logger.warning(
+            "ShowCreationThenDestructionAround has been deprecated in favor of Circumscribe. Please use Circumscribe instead!"
+        )
         super().__init__(mobject, rect_animation=rect_animation, **kwargs)
 
 
 class ShowCreationThenFadeAround(AnimationOnSurroundingRectangle):
+    """Deprecated. Use :class:`~.Circumscribe` instead."""
+
     def __init__(
         self,
         mobject: "Mobject",
         rect_animation: Animation = ShowCreationThenFadeOut,
         **kwargs
     ) -> None:
+        logger.warning(
+            "ShowCreationThenFadeAround has been deprecated in favor of Circumscribe. Please use Circumscribe instead!"
+        )
         super().__init__(mobject, rect_animation=rect_animation, **kwargs)
 
 
 class ApplyWave(Homotopy):
+    """Send a wave through the Mobject distorting it temporarily.
+
+    Parameters
+    ----------
+    mobject
+        The mobject to be distorted.
+    direction
+        The direction in which the wave nudges points of the shape
+    amplitude
+        The distance points of the shape get shifted
+    wave_func
+        The function defining the shape of one wave flank.
+    time_width
+        The length of the wave relative to the width of the mobject.
+    ripples
+        The number of ripples of the wave
+    run_time
+        The duration of the animation.
+
+    Examples
+    --------
+
+    .. manim:: ApplyingWaves
+
+        class ApplyingWaves(Scene):
+            def construct(self):
+                tex = Tex("WaveWaveWaveWaveWave").scale(2)
+                self.play(ApplyWave(tex))
+                self.play(ApplyWave(
+                    tex,
+                    direction=RIGHT,
+                    time_width=0.5,
+                    amplitude=0.3
+                ))
+                self.play(ApplyWave(
+                    tex,
+                    rate_func=linear,
+                    ripples=4
+                ))
+
+    """
+
     def __init__(
         self,
         mobject: "Mobject",
         direction: np.ndarray = UP,
         amplitude: float = 0.2,
-        run_time: float = 1,
+        wave_func: Callable[[float], float] = smooth,
+        time_width: float = 1,
+        ripples: int = 1,
+        run_time: float = 2,
         **kwargs
     ) -> None:
-        self.direction = direction
-        self.amplitude = amplitude
-        left_x = mobject.get_left()[0]
-        right_x = mobject.get_right()[0]
-        vect = self.amplitude * self.direction
+        x_min = mobject.get_left()[0]
+        x_max = mobject.get_right()[0]
+        vect = amplitude * normalize(direction)
+
+        def wave(t):
+            # Creates a wave with n ripples from a simple rate_func
+            # Do yourself a favor and don't try to understand it...
```

Comment:
Hmm Ok I find this really funny, but uuuh it'd be good to have only documented/clear code in the code base :/

Reply:
I added an exhaustive explanation :)